src/components/list/list.js

```diff
@@ -74,11 +77,102 @@ function mdListDirective() {
  * </hljs>
  *
  */
-function mdItemDirective() {
+function mdItemDirective($document, $log) {
+  var proxiedTypes = ['md-checkbox', 'md-switch'];
   return {
     restrict: 'E',
-    compile: function(element) {
-      element[0].setAttribute('role', 'listitem');
+    compile: function(tEl, tAttrs) {
+      // Check for proxy controls (no ng-click on parent, and a control inside)
+      var secondaryItem = tEl[0].querySelector('.md-secondary');
+      var hasProxiedElement;
+
+      if (!tAttrs.ngClick) {
+        for (var i = 0, type; type = proxiedTypes[i]; ++i) {
+          var proxyElement;
+          if (proxyElement = tEl[0].querySelector(type)) {
+            hasProxiedElement = true;
+            proxyElement.setAttribute('tabindex', '-1');
```

Comment:
This causes problems on checkboxes and switches, because those elements have to be focused in order for their semantics to be announced. They also shouldn't be nested in `role=button`...can we just get rid of `<div role=button>` if it's a switch?